File: app/src/main/java/com/example/obd/Demo/ScanBle.java
package com.example.obd.Demo;

import android.bluetooth.BluetoothDevice;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.View;

import com.example.obd.Demo.Adapter.SelectBle;
import com.example.obd.Demo.blelibrary.EventBus.SerchDevice;
import com.orange.obd.R;

import org.greenrobot.eventbus.EventBus;
import org.greenrobot.eventbus.Subscribe;
import org.greenrobot.eventbus.ThreadMode;

import java.util.ArrayList;

import static com.example.obd.Demo.MainActivity.scan;

public class ScanBle extends AppCompatActivity {
    ArrayList<BluetoothDevice> ble=new ArrayList<>();
    SelectBle selectBle=new SelectBle(ble,this);
    RecyclerView re;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_scan_ble);
        re=findViewById(R.id.re);
        re.setLayoutManager(new LinearLayoutManager(this));
        re.setAdapter(selectBle);
        EventBus.getDefault().register(this);
    }
    @Subscribe(threadMode = ThreadMode.MAIN)
    public void Event(SerchDevice a){
        try{
            if(!ble.contains(a.getDevic())){  ble.add(a.getDevic());
                selectBle.notifyDataSetChanged();}
        }catch (Exception e){
            Log.w("error",e.getMessage());}
    }
    public void close(View view){
        scan.scanLeDevice(false);
        this.finish();
    }
}

File: Lib/src/main/java/com/orange/blelibrary/blelibrary/Ble/ScanBle.kt
package com.orange.blelibrary.blelibrary.Ble

import android.app.Dialog
import android.bluetooth.BluetoothDevice
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import android.util.Log
import android.view.View
import android.widget.Toast

import com.orange.blelibrary.R
import com.orange.blelibrary.blelibrary.CallBack.Ble_Callback_C
import com.orange.blelibrary.blelibrary.CallBack.Dailog_SetUp_C
import com.orange.blelibrary.blelibrary.RootActivity
import com.orange.blelibrary.blelibrary.Server.ScanDevice
import kotlinx.android.synthetic.main.activity_scan_ble.*

import org.greenrobot.eventbus.Subscribe
import org.greenrobot.eventbus.ThreadMode

import java.util.ArrayList


class ScanBle(val caller:Ble_Callback_C) : Dailog_SetUp_C() {

    internal var ble = ArrayList<BluetoothDevice>()
    internal var selectBle = SelectBle(ble, this)
    var scan = ScanDevice()
    lateinit var act: RootActivity

    override fun SetUP(root: Dialog, act: RootActivity) {
        this.act=act
        root.re.layoutManager = LinearLayoutManager(act)
        root.re.adapter = selectBle
        scan.scanBle=this
        scan.setmBluetoothAdapter(act)
        root.close.setOnClickListener {
            act.DaiLogDismiss()
        }
    }

    override fun Dismiss() {
        scan.scanLeDevice(false)
        super.Dismiss()
    }

    fun DataRefresh(a: BluetoothDevice) {
        try {
            if (!ble.contains(a) && a.name != null) {
                ble.add(a)
                selectBle.notifyDataSetChanged()
            }
        } catch (e: Exception) {
            Log.w("error", e.message)
        }

    }


}
